src/message_pool.zig

```diff
@@ -108,7 +108,7 @@ pub const MessagePool = struct {
                 const message = try allocator.create(Message);
                 message.* = .{
                     .header = mem.bytesAsValue(Header, buffer[0..@sizeOf(Header)]),
-                    .buffer = buffer,
+                    .buffer = buffer[0..constants.message_size_max],
```

Comment:
Went back and forth on how to best handle this. Ideally, we'd use `allocator.createAdvanced`, but there's no such thing.

Reply:
Agreed. (In this case it might be worse b/c of the cast in deinit, but fyi another option is to allocate a single-element slice, where the element is the array.)

Reply:
yeah, single-element-slice option was uglier than this